code/modules/clothing/neck/_neck.dm

```diff
@@ -291,3 +291,110 @@
 	desc = "A sizable white cape with gold connects."
 	icon_state = "grandadmiral"
 	item_state = "grand_admiral"
+
+/*
+ * Bodycamera stuff
+ * NOTE: Unlike regular cameras, bodycams can only stream to one camera network at a time. Unless you want to give it more networks,
+ * the code will only ever worry about bodcam.network[1]
+ */
+
+/obj/item/clothing/neck/bodycam
+	name = "body camera"
+	desc = "A wearable camera, capable of streaming a live feed."
+	icon_state = "bodycam_off"
+	item_state = "bodycam_off"
+	var/prefix = "sec"//used for sprites, miner etc
+	var/obj/machinery/camera/bodcam = null
+	var/setup = FALSE
+	var/preset = FALSE //if true, the camera is already configured and cannot be reset
+	actions_types = list(/datum/action/item_action/toggle_bodycam)
+	strip_delay = 10 //takes one second to strip, so a downed officer can be un-cammed quickly
+	w_class = WEIGHT_CLASS_BULKY //you HAVE to wear it or carry it. No cheating by putting it in a bag!
+
+/obj/item/clothing/neck/bodycam/Initialize()
+	..()
+	bodcam = new(src)
+	bodcam.c_tag = "NT_BodyCam"
+	bodcam.network = list("ss13")
+	bodcam.internal_light = FALSE
+	bodcam.status = FALSE
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/attack_self(mob/user)
+	if(!setup)
+		src.AltClick(user)
+		return
+	if(bodcam.status)
+		bodcam.status = FALSE
+		to_chat(user, "<span class='notice'>You shut off the body camera.</span>")
+	else
+		bodcam.status = TRUE
+		to_chat(user, "<span class='notice'>You turn on the body camera.</span>")
+	update_icon()
+
+/obj/item/clothing/neck/bodycam/AltClick(mob/user)
+	if(preset)
+		return //can't change the settings on it if it's preset
+	var/name = input(user, "What would you like your camera's display name to be?", "Camera id", "[user.name]") as null|text
+	if(name)
+		bodcam.c_tag = "(Bodycam) " + name
+	bodcam.network[1] = input(user, "Which network should the camera broadcast to?\nFor example, 'ss13', 'security', and 'mine' are existing networks", "Camera network", "ss13") as null|text
+	if(bodcam.c_tag && bodcam.network.len > 0)
+		setup = TRUE
+		bodcam.status = TRUE
+		update_icon()
+
+/obj/item/clothing/neck/bodycam/update_icon()
+	..()
+	var/suffix = "off"
+	if (bodcam.status)
+		suffix = "on"
+	icon_state = "[prefix]_bodycam_[suffix]"
+	item_state = "[prefix]_bodycam_[suffix]"
```

Comment:
Update icons for action buttons
```suggestion
	item_state = "[prefix]_bodycam_[suffix]"
	for(var/X in actions)
		var/datum/action/A = X
		A.UpdateButtonIcon()
```

Reply:
Omg I was looking for this! Thank you so much!!!